bin/git_functions.sh

```diff
@@ -589,9 +593,15 @@ check_in_changes () {
     if [ ${CONTINUE:-y} == "Y" ] || [ ${CONTINUE:-y} == "y" ]; then
         log_term 1 "\nChecking in changes to '${_FILE}' for \"${_REPOSITORY}\"." -e
         log_term 3 "File: '${BASH_SOURCE[0]}' Line: '${LINENO}'"
-        cd "${REPOSITORY_PATH}/${_REPOSITORY}" && git add ${_FILE}
-        cd "${REPOSITORY_PATH}/${_REPOSITORY}" && git commit -m "${_MESSAGE} [skip ci]"
-        cd "${REPOSITORY_PATH}/${_REPOSITORY}" && git push
+        if ! git diff-files --quiet --ignore-submodules --; then
+            # git diff-files --name-status -r --ignore-submodules --
+            cd "${REPOSITORY_PATH}/${_REPOSITORY}" && git add ${_FILE}
```

Comment:
Note, one can use
```
git -C "${REPOSITORY_PATH}/${_REPOSITORY}" add ${_FILE}
```

Instead of
```
cd "${REPOSITORY_PATH}/${_REPOSITORY}" && git ...
```

Reply:
So noted.